CHANGELOG.md

```diff
@@ -15,6 +15,7 @@ The format is loosely based on [Keep a Changelog](http://keepachangelog.com/en/1
 
 * More precise `out` parameter detection for mocking COM interfaces with `[in,out]` parameters (@koutinho, #645)
 * Prevent false 'Different number of parameters' error with `Returns` callback methods that have been compiled from `Expression`s (@stakx, #654)
+* Align `Returns` with `Callback` method signature matching by being permissive about when `HasCompatibleParameterList` returned true (@idigra, #654).
```

Comment:
The target audience of the changelog are Moq users. They likely have no clue about an internal method like `HasCompatibleParameterList`. This would have to be rephrased to be more readily understandable for users. Heck, I know Moq internals quite well, but I'm still not sure what change we're really making here!